lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @fileoverview forbid the use of hooks from ancestor modules
+ * @author Raymond Cohen
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const NESTABLE_HOOK_NAMES = ["afterEach", "beforeEach"];
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "forbid the use of hooks from ancestor modules",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        messages: {
+            noHooksFromAncestorModules: "Do not use hooks from an ancestor module."
```

Comment:
This message is fine, but since you're passing a context, you might as well use it here!

I would suggest using the full text of the hook call (use `context.getSourceCode().getText()` on the callee node of the CallExpression) and use a message similar to the following: `"{{hook}} is from an ancestor module."`

What do you think?